app/repositories/concerns/repository.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require 'active_support/concern'
+
+module Repository
```

Comment:
Not in the slightest objecting to moving this code here; it's good. But as a point of information, do you have an immediate use for the Concern other that the place you pulled it from?

Reply:
It's included by both the `CollectionRepository` and `DocumentRepository` classes.